cloudwatch_metrics_collector/src/metrics_collector_parameter_helper.cpp

```diff
@@ -161,16 +163,42 @@ void ReadStorageResolution(
 
 void ReadTopics(
         std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
-        std::vector<std::string> & topics) {
+        std::vector<TopicInfo> & topics) {
+
+  std::vector<std::string> topic_names;
+  parameter_reader->ReadParam(ParameterPath(kNodeParamMonitorTopicsListKey), topic_names);
 
-  std::string param_key;
-  if (AWS_ERR_OK != parameter_reader->ReadParam(ParameterPath(kNodeParamMonitorTopicsListKey), param_key)) {
-    parameter_reader->ReadParam(ParameterPath(param_key), topics);
+  std::vector<int64_t> topic_types;
+  parameter_reader->ReadParam(ParameterPath(kNodeParamMonitorTopicTypesListKey), topic_names);
+
+  if (topic_types.size() != topic_names.size()) {
+    AWS_LOGSTREAM_ERROR(__func__, "Number of topic names (" << topic_names.size()
+      << ") and topic types (" << topic_types.size() << ") provided don't match");
+    return;
   }
-  if (topics.empty()) {
-    AWS_LOGSTREAM_INFO(
-            __func__, "Topic list not defined or empty. Listening on topic: " << kNodeDefaulMetricsTopic);
-    topics.push_back(kNodeDefaulMetricsTopic);
+  
+  if (topic_names.empty()) {
+    std::string info_msg("Topic list not defined or empty. Listening on topics:");
+    for (const TopicInfo & topic_info : kNodeDefaultMetricsTopics) {
+      topics.push_back(topic_info);
+      info_msg += ' ';
+      info_msg += topic_info.topic_name;
```

Comment:
mixing logging and non-logging logic in a loop is not preferable. Consider building `topics` first in a loop, then log `topics` content using another loop (preferable located in a helper function).

Reply:
No longer applicable.